File: app/src/main/java/com/luizroseiro/testeandroidv2/mainScreen/StatementsAdapter.java
package com.luizroseiro.testeandroidv2.mainScreen;

import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.support.v4.math.MathUtils;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.luizroseiro.testeandroidv2.MainActivity;
import com.luizroseiro.testeandroidv2.R;
import com.luizroseiro.testeandroidv2.models.StatementModel;

import java.math.MathContext;
import java.util.List;

public class StatementsAdapter extends RecyclerView.Adapter<StatementsViewHolder> {

    private List<StatementModel> statements;

    StatementsAdapter(List<StatementModel> statements) {
        this.statements = statements;
    }

    @NonNull
    @Override
    public StatementsViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_statement,
                viewGroup, false);
        return new StatementsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull StatementsViewHolder statementsViewHolder, int i) {
        statementsViewHolder.setStatement(statements.get(i));
    }

    @Override
    public int getItemCount() {
        return statements == null ? 0 : statements.size();
    }

}

class StatementsViewHolder extends RecyclerView.ViewHolder {

    private TextView tvTitle;
    private TextView tvDesc;
    private TextView tvDate;
    private TextView tvValue;

    StatementsViewHolder(@NonNull View itemView) {
        super(itemView);

        tvTitle = itemView.findViewById(R.id.tv_title);
        tvDesc = itemView.findViewById(R.id.tv_desc);
        tvDate = itemView.findViewById(R.id.tv_date);
        tvValue = itemView.findViewById(R.id.tv_value);
    }

    void setStatement(StatementModel statement) {
        if (statement.getValue() < 0)
            tvValue.setTextColor(ContextCompat.getColor(MainActivity.getContext(),
                    R.color.colorNegativeValue));
        else
            tvValue.setTextColor(ContextCompat.getColor(MainActivity.getContext(),
                    R.color.colorPositiveValue));

        tvTitle.setText(statement.getTitle());
        tvDesc.setText(statement.getDesc());
        tvDate.setText(statement.getDate());
        tvValue.setText(MainActivity.getContext().getString(R.string.balance_value,
                Math.abs(statement.getValue())));
    }

}

File: App/app/src/main/java/com/example/testeandroidv2/presentation/StatementsAdapter.kt
package com.example.testeandroidv2.presentation

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.testeandroidv2.R
import com.example.testeandroidv2.model.Statement
import com.example.testeandroidv2.util.currency
import com.example.testeandroidv2.util.dateFormat
import kotlinx.android.synthetic.main.item.view.*

class StatementsAdapter (
    val statements: List<Statement>
) : RecyclerView.Adapter<StatementsAdapter.StatementsViewHolder>() {

    class StatementsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val title = itemView.textTitle
        private val value = itemView.textValue
        private val date = itemView.textDate
        private val description = itemView.textDescription
        fun bindView(statment: Statement){
            title.text = statment.title
            value.text = currency(statment.value)
            date.text = dateFormat(statment.date)
            description.text = statment.desc
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StatementsViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false)
        return StatementsViewHolder(view)
    }

    override fun getItemCount() = statements.count()

    override fun onBindViewHolder(holder: StatementsViewHolder, position: Int) {
        holder.bindView(statements[position])
    }
}